### ISSUE TITLE
[Publications] 500 error User:getID() type return error

### ISSUE THREAD

**Issue by christinerogers** [aces/Loris#4562]
[Test Plan](https://github.com/aces/Loris/blob/21.0-release/modules/publication/test/TestPlan.md) - Issues found: 
- [x] 500 error on due to User:getID() type return error. resolved by:  #4575

**Comment by ridz1208** [comment]
```
[Tue May 14 09:17:56.293045 2019] [php7:error] [pid 21142] [client 132.216.56.93:33658] PHP Fatal error:  Uncaught TypeError: Return value of User::getId() must be of the type integer, string returned in /var/www/loris/php/libraries/User.class.inc:234\nStack trace:\n#0 /var/www/loris/modules/publication/php/publication.class.inc(45): User->getId()\n#1 /var/www/loris/php/libraries/Module.class.inc(280): LORIS\\publication\\Publication->_hasAccess(Object(User))\n#2 /var/www/loris/src/Middleware/ResponseGenerator.php(49): Module->handle(Object(Zend\\Diactoros\\ServerRequest))\n#3 /var/www/loris/src/Middleware/AuthMiddleware.php(62): LORIS\\Middleware\\ResponseGenerator->process(Object(Zend\\Diactoros\\ServerRequest), Object(LORIS\\publication\\Module))\n#4 /var/www/loris/src/Router/ModuleRouter.php(97): LORIS\\Middleware\\AuthMiddleware->process(Object(Zend\\Diactoros\\ServerRequest), Object(LORIS\\publication\\Module))\n#5 /var/www/loris/src/Router/BaseRouter.php(99): LORIS\\Router\\ModuleRouter->handle(Object(Zend\\Diactoros\\ServerRequest))\n#6 /var/www/loris/src/Middleware/ResponseGenerator.php(49): LORIS\\Router\\BaseRouter->handle(Object(Z in /var/www/loris/php/libraries/User.class.inc on line 234
```

**Comment by driusan** [comment]
Since it's hard to read that all on one line, the important part seems to be:

> PHP Fatal error:  Uncaught TypeError: Return value of User::getId() must be of the type integer, string returned in /var/www/loris/php/libraries/User.class.inc:234
> Stack trace:
> #0 /var/www/loris/modules/publication/php/publication.class.inc(45): User->getId()
> [...]

**Comment by christinerogers** [comment]
Thanks @driusan @ridz.  Sounds like the module code needs updating to handle the int.  I'll look into it and clarify the issue today.

**Comment by johnsaigle** [comment]
This line needs to be wrapped with an `intval()` or cast via `(int)`. This should resolve the issue.
https://github.com/aces/Loris/blob/major/php/libraries/User.class.inc#L234

**Comment by driusan** [comment]
Fixed by #4575
